### ISSUE TITLE
DataLoader reader thread raised an exception. 

### ISSUE THREAD

**Issue by FeiMa-REC** [PaddlePaddle/PaddleSeg#885]

**Comment by FeiMa-REC** [comment]
### 报错如下：

![图片](https://user-images.githubusercontent.com/62004245/111131866-a80d0d80-85b3-11eb-83e4-4c1ccadda589.png)

## 文件相关配置：

### 【_base_.yml】

---

batch_size: 2
iters: 80000

train_dataset:
  type: Dataset
  dataset_root: ./
  train_path: train_list.txt
  num_classes: 15
  transforms:
    - type: ResizeStepScaling
      min_scale_factor: 0.75
      max_scale_factor: 2.0
      scale_step_size: 0.25
    - type: RandomPaddingCrop
      crop_size: [1024, 1024]
    - type: RandomHorizontalFlip
    - type: RandomRotation 
    - type: RandomDistort
      brightness_range: 0.4
      contrast_range: 0.4
      saturation_range: 0.4
    - type: Normalize
  mode: train

val_dataset:
  type: Dataset
  dataset_root: ./
  val_path: val_list.txt
  num_classes: 15
  transforms:
    - type: RandomHorizontalFlip
    - type: Normalize
  mode: val

optimizer:
  type: sgd
  momentum: 0.9
  weight_decay: 4.0e-4

learning_rate:
  value: 0.0075
  decay:
    type: poly
    power: 0.9
    end_lr: 0.0

loss:
  types:
    - type: MixedLoss
      losses:
        - type: CrossEntropyLoss
        - type: LovaszSoftmaxLoss
      coef: [0.8, 0.2]
    - type: MixedLoss
      losses:
        - type: CrossEntropyLoss
        - type: LovaszSoftmaxLoss
      coef: [0.8, 0.2]
  coef: [1, 0.4]

### 【train.yml】

---

_base_: '../_base_/SmartCar_base_Config.yml'

batch_size: 2
iters: 80000

model:
  type: DeepLabV3P
  backbone:
    type: ResNet101_vd
    output_stride: 8
    multi_grid: [1, 2, 4]
    pretrained: https://bj.bcebos.com/paddleseg/dygraph/resnet101_vd_ssld.tar.gz
  num_classes: 15
  backbone_indices: [0, 3]
  aspp_ratios: [1, 12, 24, 36]
  aspp_out_channels: 256
  align_corners: False
  pretrained: null

### 【文件结构】：

---

![图片](https://user-images.githubusercontent.com/62004245/111132421-4ac58c00-85b4-11eb-9aab-f5fe60fe4202.png)

![图片](https://user-images.githubusercontent.com/62004245/111132435-4f8a4000-85b4-11eb-98c2-15c93f68fb3e.png)


### 【train_list】：

---

![图片](https://user-images.githubusercontent.com/62004245/111132477-5f098900-85b4-11eb-94f3-e310cb611bab.png)

**Comment by BeyondYourself** [comment]
预训练模型直接用压缩文件？？ 都不解压么

**Comment by FeiMa-REC** [comment]
> 预训练模型直接用压缩文件？？ 都不解压么

执行训练流程的时候会在下载完预训练模型后自动解压吧？我看到有解压步骤的出现的。这个应该是dataloader的问题`(*>﹏<*)′

**Comment by michaelowenliu** [comment]
@YuBCS-Ma I guess data_root is wrong.

**Comment by FeiMa-REC** [comment]
> @michaelowenliu  I guess data_root is wrong.

### My file tree is as follows :

![图片](https://user-images.githubusercontent.com/62004245/111461725-8ac57400-8758-11eb-80d1-10974933de2e.png)


### && My Config root is : 

train_dataset:
  type: Dataset
  dataset_root: data/
  train_path: data/train_list.txt

val_dataset:
  type: Dataset
  dataset_root: data/
  val_path: data/val_list.txt

### My train_list.txt : 

![图片](https://user-images.githubusercontent.com/62004245/111462429-65853580-8759-11eb-8b09-92355c7fa6c0.png)


## But this time it still reported a mistake : <


![图片](https://user-images.githubusercontent.com/62004245/111462820-d593bb80-8759-11eb-8002-1f2928a88adc.png)
 
![图片](https://user-images.githubusercontent.com/62004245/111462829-d7f61580-8759-11eb-9cd8-263388170506.png)

**Comment by BeyondYourself** [comment]
![image](https://user-images.githubusercontent.com/12640462/111463991-2eb01f00-875b-11eb-9b09-a28084964b6d.png)
你在配置文件中添加上transform
设置它的属性Resize试试
@YuBCS-Ma

**Comment by FeiMa-REC** [comment]
> 在配置文件中添加上transform
> 设置它的属性Resize试试
> @BeyondYourself 

啊，老哥，那个错误我解决了。resize没事，就是一直报数据读取的错。但是我的文件路径都对啊，不知道为什么跑不通。

![图片](https://user-images.githubusercontent.com/62004245/111464565-dfb6b980-875b-11eb-9417-e01df439517a.png)

刚刚跑的。

**Comment by michaelowenliu** [comment]
@YuBCS-Ma how did you run "train.py"? what is the absolute path of the data directory?

**Comment by FeiMa-REC** [comment]
> @michaelowenliu  how did you run "train.py"? what is the absolute path of the data directory?

In this way : 

![图片](https://user-images.githubusercontent.com/62004245/111467291-3c67a380-875f-11eb-87ea-b2611f677788.png)


&& The absolute path of the data is : home/aistudio/data/ 

![图片](https://user-images.githubusercontent.com/62004245/111467968-0545c200-8760-11eb-88c9-addfb1830f7e.png)

**Comment by BeyondYourself** [comment]
我之前遇到过报错和你一模一样的，结果是transform的属性值配置顺序有问题，resize最为最后属性值 @YuBCS-Ma

**Comment by FeiMa-REC** [comment]
> 我之前遇到过报错和你一模一样的，结果是transform的属性值配置顺序有问题，resize最为最后属性值 @BeyondYourself 

这是我的Transform

![图片](https://user-images.githubusercontent.com/62004245/111471242-abdf9200-8763-11eb-84f2-42026392e09f.png)

**Comment by BeyondYourself** [comment]
你先试试属性值只加上Resize和Normalize  @YuBCS-Ma

**Comment by FeiMa-REC** [comment]
> 你先试试属性值只加上Resize和Normalize @BeyondYourself 

我试了只加Normalize，还是一样的报错，应该不是这个的问题。

**Comment by BeyondYourself** [comment]
这种报错有的时候真的不太好找，建议debug看看

**Comment by michaelowenliu** [comment]
@YuBCS-Ma Hi, please try running train.py in PaddleSeg-Release-v2.0 directory, and set data_root to home/aistudio/data/, set train_path: home/aistudio/data/train_list.txt.

**Comment by BeyondYourself** [comment]
@michaelowenliu 所以在读取文件是不是应该加一个验证文件以及路径是否存在

**Comment by michaelowenliu** [comment]
@BeyondYourself Hi, a great idea! Hope you can help to add this feature and submit the PR.

**Comment by BeyondYourself** [comment]
OK，我试试看

**Comment by wuyefeilin** [comment]
> > 我之前遇到过报错和你一模一样的，结果是transform的属性值配置顺序有问题，resize最为最后属性值 @BeyondYourself
> 
> 这是我的Transform
> 
> ![图片](https://user-images.githubusercontent.com/62004245/111471242-abdf9200-8763-11eb-84f2-42026392e09f.png)

RandomRotation 会改变图片大小，把它放到RandomPaddingCrop之前

**Comment by BeyondYourself** [comment]
@wuyefeilin 出个transform的配置宝典吧

**Comment by wuyefeilin** [comment]
> @wuyefeilin 出个transform的配置宝典吧

感谢反馈，我们会充分考虑你的建议

**Comment by FeiMa-REC** [comment]
@michaelowenliu @wuyefeilin 

Finally, it works！！！：）

I should set_env PYTHONPATH='/home/aistudio/PaddleSeg-release-v2.0/' and change dataset_root , train_path, also should  put the RandomRotation before the RandomPaddingCrop.

Thx! ! !

**Comment by Exusia-0** [comment]
> > > 我之前遇到过报错和你一模一样的，结果是transform的属性值配置顺序有问题，resize最最后的属性值@BeyondYourself
> > 
> > 
> > 这是我的Transform
> > ![图片](https://user-images.githubusercontent.com/62004245/111471242-abdf9200-8763-11eb-84f2-42026392e09f.png)
> 
> RandomRotation会改变图片大小，把它放到RandomPaddingCrop之前

确实是这个问题
